packages/frontend/src/pages/api/v1/posts.ts

```diff
@@ -45,25 +48,34 @@ const handleGetPosts = async (req: NextApiRequest, res: NextApiResponse<IPostPre
 
 // Handle non-pseudonymous post creation
 // Verify the ECDSA signature and save the post
-const handleCreateDoxedPost = async (req: NextApiRequest, res: NextApiResponse) => {
+const handleCreateDoxedPost = async (
+  req: NextApiRequest,
+  res: NextApiResponse<{} | { error: string }>,
+) => {
   const content: Content = req.body.content;
   const sig: string = req.body.attestation;
 
   if (!isTimestampValid(content.timestamp)) {
-    res.status(400).send('Invalid timestamp!');
+    res.status(400).send({ error: INVALID_TIMESTAMP });
     return;
   }
 
   if (!verifyRoot(content.groupRoot)) {
-    res.status(400).send('Invalid group root!');
+    res.status(400).send({ error: INVALID_MERKLE_ROOT });
     return;
   }
 
   const post = toPost(content, sig, AttestationScheme.EIP712);
-  const pubKey = recoverPostPubkey(post);
+  let pubKey;
+  try {
+    pubKey = recoverPostPubkey(post);
+  } catch (_err) {
+    res.status(400).send({ error: INVALID_SIGNATURE });
+    return;
+  }
 
   if (!(await verifyInclusion(pubKey))) {
-    res.status(400).send('Public key not in latest group');
+    res.status(400).send({ error: USER_NOT_IN_LATEST_GROUP });
     return;
   }
 
```

Comment:
@DanTehrani do we want to put the entire logic in a try/catch block too and have a generic `GET POSTS failed, ${error message}` response? (we could replicate this pattern for all API requests)

Reply:
@cha0sg0d 
Nice idea! I'll change it to a master try/catch

Reply:
don't know if it makes sense in prod, but at least for debugging purposes may make sense to return the merkle root that user was not seen in here

Reply:
@cha0sg0d @lsankar4033 
Yeah we might want to return dynamic error messages, so a top-level try/catch could complicate things. 

Can we merge in this form for now, and later update the error handling according to the needs of the frontend?